doc: amend support tier qualifier (#42805)
nodejs/node: doc/contributing/diagnostic-tooling-support-tiers.md

    organization or website;
  * The tool must be open source.

> * Tier 3 - Its test suite will be run either as nightly in Node.js CI

Reviewer: I'm ok with this but it is stronger than what was there before. It will mean that
a tool cannot be Tier 3 without testing in place.

Author: @mhdawson - agree; but isn't it what we actually desire for / state of tools that are already in tier 3? a test that is either complete or partial, configured / run either in the CI or in CITGM? If not, pls advise a better wording. For example, will this be an improvement?

`Tier 3 - If possible its test suite will be run at least nightly in the Node.js CI or in CITGM, and issues opened for failures.  Does not block shipping a release.`

Reviewer: @gireeshpunathil, the only case where it would matter is if there is a Tier 3 tool which can't be added to the CI for some reason. The previous wording would still allow it to be Tier 3. Having said that we can always adjust if that becomes an issue and in general it is better good the force adding the testing if it is possible.

Reviewer: So good with your update.

Author: adjusted the wordings.

> * Tier 3 - Its test suite will be run either as nightly in Node.js CI
>   or in CITGM (in case if a test cannot be configured and run in nightly
>   by virtue of the nature of the tool in question) and issues opened for
>   failures. Does not block shipping a release.

Reviewer 2: ```suggestion
* Tier 3 - If possible its test suite will be run at least nightly in the
  Node.js CI or in CITGM and issues opened for failures. Does not block shipping
  a release.
```
I think we can just add the `or in CITGM` part?

Author: yes, thanks. I asked the same question to @mhdawson above, if he too is fine , I will go with it.

Author: done
